instrumentation/google.golang.org/grpc/otelgrpc/test/interceptor_test.go

```diff
@@ -1069,3 +1094,22 @@ func TestStreamServerInterceptorEvents(t *testing.T) {
 		})
 	}
 }
+
+func checkManualReaderRecords(t *testing.T, reader metric.Reader, serviceName, name string, code grpc_codes.Code) {
+	rm := metricdata.ResourceMetrics{}
+	err := reader.Collect(context.Background(), &rm)
+	assert.NoError(t, err)
+	require.Len(t, rm.ScopeMetrics, 1)
+	require.Len(t, rm.ScopeMetrics[0].Metrics, 1)
+	require.IsType(t, rm.ScopeMetrics[0].Metrics[0].Data, metricdata.Histogram[int64]{})
+	data := rm.ScopeMetrics[0].Metrics[0].Data.(metricdata.Histogram[int64])
+	dpt, ok := getMetricFromData(data, name)
+	assert.True(t, ok)
+	attr := dpt.Attributes.ToSlice()
+	assert.ElementsMatch(t, []attribute.KeyValue{
+		semconv.RPCMethod(name),
+		semconv.RPCService(serviceName),
+		otelgrpc.RPCSystemGRPC,
+		otelgrpc.GRPCStatusCodeKey.Int64(int64(code)),
+	}, attr)
```

Comment:
I suggest using `metricdatatest` instead. 

```suggestion
	want := metricdata.ScopeMetrics{
		Scope: instrumentation.Scope{
			Name:      "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc",
			SchemaURL: "https://opentelemetry.io/schemas/1.17.0",
			Version:   otelgrpc.Version(),
		},
		Metrics: []metricdata.Metrics{
			{
				Name:        "rpc.server.duration",
				Description: "Measures the duration of inbound RPC.",
				Unit:        "ms",
				Data: metricdata.Histogram[int64]{
					Temporality: metricdata.CumulativeTemporality,
					DataPoints: []metricdata.HistogramDataPoint[int64]{
						{
							Attributes: attribute.NewSet(
								semconv.RPCMethod(name),
								semconv.RPCService(serviceName),
								otelgrpc.RPCSystemGRPC,
								otelgrpc.GRPCStatusCodeKey.Int64(int64(code)),
							),
						},
					},
				},
			},
		},
	}

	rm := metricdata.ResourceMetrics{}
	err := reader.Collect(context.Background(), &rm)
	assert.NoError(t, err)
	require.Len(t, rm.ScopeMetrics, 1)
	metricdatatest.AssertEqual(t, want, rm.ScopeMetrics[0], metricdatatest.IgnoreTimestamp(), metricdatatest.IgnoreValue())
```

To make this working you would also need to change the `TestUnaryServerInterceptor` to update to setup providers, recorders etc for each test case. I also noticed that the providers are not being shutdown.

I think it is better to address it in a separate PR as I think this refactoring could be done for most of the gRPC tests.

Reply:
I created https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4483

Reply:
Aha, I see, thanks, as you probably noticed I basically copied such comparison code from other places.

So it would be nice if we could leave this as a following up refactory task.

However, for now, I cannot promise I have more free time to do such following-up fixes. Hopefully, that is okay?